subsys/mgmt/mcumgr/grp/os_mgmt_client/src/os_mgmt_client.c

```diff
@@ -138,7 +138,8 @@ int os_mgmt_client_echo(struct os_mgmt_client *client, const char *echo_string)
 	zcbor_new_encode_state(zse, ARRAY_SIZE(zse), nb->data + nb->len, net_buf_tailroom(nb), 0);
 
 	ok = zcbor_map_start_encode(zse, 2) &&
-	     zcbor_tstr_put_lit(zse, "d") && zcbor_tstr_put_term(zse, echo_string) &&
+	     zcbor_tstr_put_lit(zse, "d") &&
+	     zcbor_tstr_put_term(zse, echo_string, CONFIG_ZCBOR_MAX_STR_LEN) &&
```

Comment:
@nordicjm I agree. I managed to limit it to only be used in tests, and I changed it to be 256 instead of 4096. And I added help text discouraging its use.